sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JSONOptions.scala

```diff
@@ -136,4 +138,6 @@ private[sql] class JSONOptions(
       allowBackslashEscapingAnyCharacter)
     factory.configure(JsonParser.Feature.ALLOW_UNQUOTED_CONTROL_CHARS, allowUnquotedControlChars)
   }
+
+  val textOptions = ListMap(parameters.toList: _*)
```

Comment:
I think we can use `parameters` instead of constructing a new map.

Reply:
The reason why I constructed new map is I got an exception like the `textOptions` value is not serializable on one test suite.

Reply:
I see, `CaseInsensitiveMap.. with Serializable` so I think it was ok. I didn't try it and run tests.

Reply:
I faced to the problem on `FileStreamSourceSuite.read new files in partitioned table with globbing, should not read partition data`. Let me check it again.

Reply:
The test fails on serialization of the `textOptions` value because the former one refers to `transient` parameters. I add the `@transient` annotation to `textOptions` (it should be safe because `textOptions` is using on the driver only and shouldn't be serialized).